File: 8-kyu-kata/counting-sheep-dot-dot-dot/python/counting_sheep_dot_dot_dot_test.py
# 8 kyu - Counting sheep...  [ ID: 54edbc7200b811e956000556  (counting-sheep-dot-dot-dot) ]
# URL: https://www.codewars.com/kata/54edbc7200b811e956000556
# Category: REFERENCE  |  Tags: ARRAYS | FUNDAMENTALS
# ******************************************************************************
array1 = [
    True,
    True,
    True,
    False,
    True,
    True,
    True,
    True,
    True,
    False,
    True,
    False,
    True,
    False,
    False,
    True,
    True,
    True,
    True,
    True,
    False,
    False,
    True,
    True,
]

array2 = []
array2.extend([True] * 500)
array2.extend([None] * 5)
array2.extend([False] * 100)

array3 = []

test.assert_equals(result := count_sheeps(array1), 17, "There are 17 sheeps in total, not %s" % result)
test.assert_equals(result := count_sheeps(array2), 500, "There are 500 sheeps in total, not %s" % result)
test.assert_equals(result := count_sheeps(array3), 0, "There are no sheeps at all, you counted %s" % result)

File: 8-kyu-kata/counting-sheep-dot-dot-dot/javascript/countingSheepDotDotDot.Test.js
// 8 kyu - Counting sheep...  [ ID: 54edbc7200b811e956000556  (counting-sheep-dot-dot-dot) ]
// URL: https://www.codewars.com/kata/54edbc7200b811e956000556
// Category: REFERENCE  |  Tags: ARRAYS | FUNDAMENTALS
// *****************************************************************************
describe("Tests", () => {
  it("test", () => {
let array1 = [true,  true,  true,  false,
              true,  true,  true,  true ,
              true,  false, true,  false,
              true,  false, false, true ,
              true,  true,  true,  true ,
              false, false, true,  true ];
            
let array2 = [];
for (let index = 0; index < 500; index++)
  array2.push(true);
for (let index = 0; index < 5; index++)
  array2.push(undefined);
for (let index = 0; index < 100; index++)
  array2.push(false);
              
let array3 = [];
for (let index = 0; index < 500; index++)
  array3.push(null);
for (let index = 0; index < 5; index++)
  array3.push(undefined);
for (let index = 0; index < 100; index++)
  array3.push(false);
  
Test.assertEquals(countSheeps(array1), 17, "There are 17 sheeps in total, not " + countSheeps(array1))
Test.assertEquals(countSheeps(array2), 500, "There are 500 sheeps in total, not " + countSheeps(array2))
Test.assertEquals(countSheeps(array3), 0, "There are no sheeps at all, you counted " + countSheeps(array3))
  });
});

